pkg/cmd/registry/artifacts/download/download.go

```diff
@@ -0,0 +1,170 @@
+	opts := &Options{
+		Config:     f.Config,
+		Connection: f.Connection,
+		IO:         f.IOStreams,
+		localizer:  f.Localizer,
+		Logger:     f.Logger,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "get",
+		Short: "Download artifacts from registry by using global identifiers",
+		Long: `Get single or more artifacts by group, content, hash or globalIds. 
+		NOTE: Use "service-registry get" command if you wish to download artifact by artifactId.
+
+		Flags are used to specify the artifact to download:
+
+		--contentId - id if the content from metadata
+		--globalId - globalId of the content from metadata
+		--hash - SHA-256 hash of the content`,
+		Example: `
+## Get latest artifact by content id
+rhoas service-registry artifacts download --contentId=183282932983
+
+## Get latest artifact by content id to specific file
+rhoas service-registry artifacts download --contentId=183282932983 schema.json
+
+## Get latest artifact by global id
+rhoas service-registry artifacts download --globalId=383282932983
+
+## Get latest artifact by hash
+rhoas service-registry artifacts download --hash=c71d239df91726fc519c6eb72d318ec65820627232b2f796219e87dcf35d0ab4
+`,
+		Args: cobra.RangeArgs(0, 1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) > 0 {
+				opts.outputFile = args[0]
+			}
+
+			if opts.registryID != "" {
+				return runGet(opts)
+			}
+
+			cfg, err := opts.Config.Load()
+			if err != nil {
+				return err
+			}
+
+			if !cfg.HasServiceRegistry() {
+				return fmt.Errorf("No service Registry selected. Use 'rhoas service-registry use' to select your registry")
+			}
+
+			opts.registryID = fmt.Sprint(cfg.Services.ServiceRegistry.InstanceID)
+			return runGet(opts)
+		},
+	}
+
+	cmd.Flags().StringVarP(&opts.group, "group", "g", "", "Group of the artifact")
+	cmd.Flags().StringVarP(&opts.hash, "hash", "", "", "SHA-256 hash of the artifact")
+	cmd.Flags().Int64VarP(&opts.globalId, "globalId", "", -1, "Global ID of the artifact")
+	cmd.Flags().Int64VarP(&opts.contentId, "contentId", "", -1, "contentId of the artifact")
+
+	cmd.Flags().StringVarP(&opts.outputFile, "outputFile", "", "", "Filename of the output file")
+
+	cmd.Flags().StringVarP(&opts.registryID, "registryId", "", "", "Id of the registry to be used. By default uses currently selected registry")
+
+	flagutil.EnableOutputFlagCompletion(cmd)
+
+	return cmd
+}
+
+func runGet(opts *Options) error {
+	logger, err := opts.Logger()
+	if err != nil {
+		return err
+	}
+
+	conn, err := opts.Connection(connection.DefaultConfigRequireMasAuth)
+	if err != nil {
+		return err
+	}
+
+	dataAPI, _, err := conn.API().ServiceRegistryInstance(opts.registryID)
+	if err != nil {
+		return err
+	}
+
+	if opts.group == "" {
+		logger.Info("Group was not specified. Using 'default' artifacts group.")
+		opts.group = util.DefaultArtifactGroup
+	}
+
+	logger.Info("Fetching artifact")
+
+	ctx := context.Background()
+	var dataFile *os.File
+	// nolint
+	if opts.contentId != -1 {
```

Comment:
What does `-1` mean? We should create a constant for readbility.

Reply:
Actually rookie mistake. Variables in flag should be *int64 and then we can check for nil as default value

Reply:
I don't think that is possible.